pandas/core/dtypes/cast.py

```diff
@@ -1654,12 +1654,33 @@ def cast_scalar_to_array(
 
     """
     if dtype is None:
-        dtype, fill_value = infer_dtype_from_scalar(value)
+        dtype, value = infer_dtype_from_scalar(value)
     else:
-        fill_value = value
+        if not isinstance(dtype, np.dtype):
+            dtype = np.dtype(dtype)
+        empty = shape and not any(shape)
+        # rem: type coercion if empty: sometimes yes, sometimes no ?
+
+        if not empty and is_integer_dtype(dtype) and isna(value):
+            # coerce if we have nan for an integer dtype
+            dtype = np.dtype("float64")
+        elif isinstance(dtype, np.dtype) and dtype.kind in ("U", "S"):
+            # we need to coerce to object dtype to avoid
+            # to allow numpy to take our string as a scalar value
+            dtype = np.dtype("object")
+            if not isna(value):
+                value = ensure_str(value)
+        elif dtype.kind in ["M", "m"]:
+            # GH38032: filling in Timedelta/Timestamp drops nanoseconds
+            if isinstance(value, (Timedelta, Timestamp)):
```

Comment:
this is the change yes?

Reply:
this will be lossy for tzaware

Reply:
Thank you both for your comments. Yes this is the change.

At first I went to update `construct_1d_arraylike_from_scalar`, but tests showed more was broken.

`cast_scalar_to_array` is used in DF constructor and but did not check its dtype arg like `construct_1d_arraylike_from_scalar`,  hence e.g.
```
In [2]: pd.DataFrame("hello", index=[0], columns=[0], dtype="U")                                                                                                                                                                           
Out[2]: 
   0
0  h
```

Reply:
I wanted to isolate the part taking care of converting a given (scalar, np.dtype) into smth suitable for creating a ndarray filled with that scalar, and `cast_scalar_to_array` felt like a good place.